WatchApp Extension/Scenes/GlucoseChartScene.swift

```diff
@@ -250,32 +249,51 @@ class GlucoseChartScene: SKScene {
         // Make temporary overrides visually match what we do in the Loop app. This means that we have
         // one darker box which represents the duration of the override, but we have a second lighter box which
         // extends to the end of the visible window.
-        if let range = activeOverride, let rangeHashable = TemporaryScheduleOverrideHashable(range) {
-            let (sprite1, created) = getSprite(forHash: rangeHashable.chartHashValue)
+        func plotOverride(
+            _ override: TemporaryScheduleOverride,
+            pushingStartTo startDate: Date? = nil,
+            extendingToChartEnd shouldExtendToChartEnd: Bool
+        ) {
+            var override = override
+            if let startDate = startDate {
+                guard startDate < override.endDate else {
```

Comment:
I'm confused about this check. Can you explain?

Reply:
`startDate` is the initial point for which the override should be plotted. If the override ends before `startDate`, it should not be plotted.

In context, if pre-meal is active in conjunction with another override, the override should only be plotted during the period for which it applies, i.e. after pre-meal ends.